pkg/impersonation/impersonation.go

```diff
@@ -361,24 +361,45 @@ func (i *Impersonator) getUser(userInfo user.Info) (user.Info, error) {
 	}
 
 	groups := []string{"system:authenticated", "system:cattle:authenticated"}
+	extras := make(map[string][]string)
 	attribs, err := i.userAttributeLister.Get("", userInfo.GetUID())
 	if err != nil && !apierrors.IsNotFound(err) {
 		return &user.DefaultInfo{}, err
 	}
-	if attribs != nil {
+	if attribs == nil { // system users do not have userattributes, but principalid and username are on the user
+		if u.Username != "" {
+			extras["username"] = []string{u.Username}
+		} else if u.DisplayName != "" {
+			extras["username"] = []string{u.DisplayName}
+		}
+		if len(u.PrincipalIDs) > 0 {
+			extras["principalid"] = u.PrincipalIDs
```

Comment:
Small Item - In light of [#35325](https://github.com/rancher/rancher/pull/35325), would you mind changing the hardcoded strings here (and in L371, L373, L387, L388, L391, L393, etc.) to use a constant? I see that pkg/auth/providers/common/provider.go has two for these values, but you can also just define them at the top of the file if you don't want to import that (or can't for some reason).

Reply:
done